cinema/views.py

```diff
@@ -1 +1,56 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import Movie, Actor, Genre, CinemaHall, MovieSession
+from cinema.serializers import (
+    ActorSerializer,
+    CinemaHallSerializer,
+    GenreSerializer,
+    MovieSerializer,
+    MovieListSerializer,
+    MovieSessionSerializer,
+    MovieSessionDetailSerializer,
+    MovieDetailSerializer,
+    MovieSessionListSerializer
+)
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class CinemaHallSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+    def get_view_name(self):
+        if "list" in self.settings.VIEW_NAME_FUNCTION(self):
+            return "Cinema Halls"
+        return "Cinema Hall Instance"
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.prefetch_related("actors", "genres")
```

Comment:
do u need optimized query in actions except list, retrieve ?

Reply:
where do I use an optimized query except list/retrieve?
I'm pretty much sure, that only serializers are working in those cases